webui/react/src/components/CreateExperimentModal.tsx

```diff
@@ -0,0 +1,88 @@
+import { Alert, Modal } from 'antd';
+import yaml from 'js-yaml';
+import React, { SetStateAction, useCallback, useState } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+
+import { routeAll } from 'routes';
+import { forkExperiment } from 'services/api';
+
+import css from './CreateExperimentModal.module.scss';
+
+export interface CreateExpModalState {
+  visible: boolean;
+  config: string;
+}
+
+interface Props {
+  title: string;
+  okText: string;
+  parentId: number;
+  visible: boolean;
+  config: string;
+  setState: (arg0: SetStateAction<CreateExpModalState>) => void;
```

Comment:
**issue:** TL;DR - avoid passing `setState` to child component and instead pass in a `setState`

The crux problem of this anti-pattern is that the state writing operation is in two places, the parent and the child component. The `on[Action]` pattern is used to keep the state and setState contained in the parent and the child modal will just send events back to the parent for it to handle the source of truth (state). It's also perfectly fine to have multiple `on[Action]` handler because the idea there is to orthogonalize unrelated events from each other, so the handlers do not have to worry about one event from another. This is especially important with the `useCallback` hook, because of the dependency list.